### PR TITLE
Upgrade typescript & shared eslint configuration

### PR DESCRIPTION
**Pull request by NSeydoux**
This upgrades to the latest version of typescript, and fixes the mismatch in the fetch function signature it involves.

### CODE REVIEW

**Review by matthieubosquet — APPROVED** [review]
Updating TypeScript is fine, but you mixed it up with https://github.com/inrupt/solid-client-access-grants-js/pull/277/files.

Maybe remove the issueAccess* changes before merging?

**Comment by NSeydoux** [comment]
Yeah #277 was required to pass e2e tests. I was planing on merging it first, and then rebase this on top.

File: package.json [hunk 1]
```diff
@@ -81,8 +81,8 @@
   "devDependencies": {
     "@inrupt/eslint-config-base": "^1.0.0",
     "@inrupt/eslint-config-lib": "^1.0.0",
```
Comment (ThisIsMissEm) [inline package.json:83]:
```suggestion
    "@inrupt/eslint-config-lib": "^1.0.0",
```

You no longer need `base` here.

Comment (ThisIsMissEm) [reply package.json:83]: Actually, ignore that. This repo hasn't been correctly upgraded to 1.0.0

**Review by ThisIsMissEm — CHANGES_REQUESTED** [review]
@NSeydoux I'm going to take this over, as we need to do the correct upgrade to the new eslint configs, so I'll push a commit with the changes required for that.

**Comment by ThisIsMissEm** [comment]
Okay, the Vercel crash here is related to the sourceurl plugin and typedoc 0.23.3, due to this breaking change:
> Removed ProjectReflection.files, this was an internal cache that should not have been exposed, and shouldn't have existed in the first place, since removing it made TypeDoc faster.

I'm going to remove the sourceurl plugin for now, as it was a "nice to have" on Vercel, but worked out of the box on the published docs.

**Review by ThisIsMissEm — APPROVED** [review]
I think this is now good to go.

**Pull request merged**